src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php

```diff
@@ -71,9 +72,15 @@ public function normalize($object, $format = null, array $context = [])
             unset($data['originalPrice']);
         }
 
+        /** @var ArrayCollection $appliedPromotions */
         $appliedPromotions = $object->getAppliedPromotionsForChannel($channel);
-        if (!empty($appliedPromotions)) {
-            $data['appliedPromotions'] = $appliedPromotions;
+        if (!$appliedPromotions->isEmpty()) {
+            $data['appliedPromotions'] = array_map(fn (CatalogPromotionInterface $catalogPromotion) => sprintf(
+                '/api/v2/shop/catalog-promotions/%s',
+                $catalogPromotion->getCode()
+            ),
+                $appliedPromotions->toArray()
+            );
```

Comment:
I would be for using here an IriConverter::getIriFromItem instead of creating a string with a hardcoded path